Assets/HoloToolkit/SpectatorView/Scripts/SpatialSync/CameraCaptureHololens.cs

```diff
@@ -0,0 +1,167 @@
+        /// </summary>
+        private Texture2D targetTexture;
+
+        /// <summary>
+        /// Starts capturing photos
+        /// </summary>
+        public void StartCapture()
+        {
+#if NETFX_CORE
+			if(!capturing)
+			{
+				PhotoCapture.CreateAsync(false, OnPhotoCaptureCreated);
+			    capturing = true;
+			}
+#else
+			Debug.LogWarning("Capturing only supported on the HoloLens platform");
+#endif
+        }
+
+        /// <summary>
+        /// Stops capturing photos
+        /// </summary>
+        public void StopCapture()
+        {
+#if NETFX_CORE
+			if(capturing)
+			{
+				photoCaptureObject.StopPhotoModeAsync(OnStoppedPhotoMode);
+			}
+			capturing = false;
+#else
+			Debug.LogWarning("Capturing only supported on the HoloLens platform");
+#endif
+        }
+
+#if NETFX_CORE
+
+        /// <summary>
+        /// Called when a capture object has been created, it configures the camera for the capture process
+        /// </summary>
+        /// <param name="captureObject">Contains the camera intent to open</param>
+        private void OnPhotoCaptureCreated(PhotoCapture captureObject)
+        {
+            photoCaptureObject = captureObject;
+
+            var cameraParameters = new CameraParameters();
+            cameraParameters.hologramOpacity = 0.0f;
+            photoHeight = 504;
+            photoWidth = 896;
+            cameraParameters.cameraResolutionWidth = photoWidth;
+            cameraParameters.cameraResolutionHeight = photoHeight;
+            cameraParameters.pixelFormat = CapturePixelFormat.BGRA32;
+            captureObject.StartPhotoModeAsync(cameraParameters, OnPhotoModeStarted);
+        }
+
+        /// <summary>
+        /// Called when the photo mode starts, if it's successfull then it'll start taking photos
+        /// </summary>
+        /// <param name="result">Result of the intent of starting the camera</param>
+        private void OnPhotoModeStarted(PhotoCapture.PhotoCaptureResult result)
+        {
+            if (result.success)
+            {
+                photoCaptureObject.TakePhotoAsync(OnCapturedPhotoToMemory);
+            }
+            else
+            {
+                Debug.LogError("Unable to start photo mode!");
+            }
+        }
+
+        /// <summary>
+        /// Called when a photo has been captured to memory, if successfull,
+        /// it'll copy the photo to the target texture
+        /// </summary>
+        /// <param name="result">Result of the photo process</param>
+        /// <param name="photoCaptureFrame">Contains the photo information</param>
+        private void OnCapturedPhotoToMemory(PhotoCapture.PhotoCaptureResult result, PhotoCaptureFrame photoCaptureFrame)
+        {
+            if (result.success)
+            {
+                if(targetTexture != null)
+                {
+                    Destroy(targetTexture);
+                }
+
+                targetTexture = new Texture2D(896, 504, TextureFormat.RGB24, false);
```

Comment:
Same Magjick numbers again.

Reply:
where do these values come from?

Reply:
This is the low power resolution the HoloLens gives you from the camera, https://docs.microsoft.com/en-us/windows/mixed-reality/locatable-camera